### PR TITLE
Adding the option for the system to use the HOME env for windows.

### PR DESCRIPTION
**Pull request by mayman**
Adding the option of checking for the HOME environment on windows. As on a work network, HOMEDRIVE is setup from the Group Policy on a work network.

Initial change was made https://github.com/kubernetes/kubernetes/pull/17590

This change would fix this issue: 
https://github.com/kubernetes/minikube/issues/459

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32790)

### CODE REVIEW

**Comment by smarterclayton** [comment]
@fabianofranz

**Comment by mayman** [comment]
@smarterclayton @fabianofranz 

Hello, 

Apologies for bugging, but would somebody be able to have a look at the pull request please.

Kind Regards,

Dilyan

File: pkg/util/homedir/homedir.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 
 // HomeDir returns the home directory for the current user
 func HomeDir() string {
-	if runtime.GOOS == "windows" {
+	if runtime.GOOS == "windows"; len(os.Getenv("HOME")) <= 0) {
```
Comment (fabianofranz) [inline pkg/util/homedir/homedir.go]: Leave this line as is and do, right below inside the windows condition:

```
if home := os.Getenv("HOME"); len(home) > 0 {
  if _, err := os.Stat(home); err == nil {
    return home
  }
}
```

on Windows we always need a `os.Stat` to make sure it's good to use.

Comment (fabianofranz) [inline pkg/util/homedir/homedir.go]: You will probably need a migration [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/clientcmd/loader.go#L56-L66) (Windows only) to make sure that users get their configs migrated correctly if they are relying on the current approach.

**Review by fabianofranz — CHANGES_REQUESTED** [review]

File: pkg/util/homedir/homedir.go (hunk 1 above)
Comment (mayman) [inline pkg/util/homedir/homedir.go]: I thought that I could do something similar as the homedir.go, but then I noticed that 
RecommendedHomeFile is also used [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/clientcmd/loader.go#L134-L149)

I am not entirely sure if this won't break anything further down the line.

Could you please have a look and let me know if you could recommend something else?
~~~Go
if goruntime.GOOS == "windows" {
    if home := os.Getenv("HOME"); len(home) > 0 {
        if _, err := os.Stat(home); err = nill {
            migrationRules[oldRecommendedWindowsHomeFile] = RecommendedHomeFile
        }
    } else {
        migrationRules[RecommendedHomeFile] = oldRecommendedWindowsHomeFile
    }
}
~~~

**Comment by mayman** [comment]
Hi @fabianofranz, could you please have a look at my latest query?

Kind regards,

Dilyan

**Comment by mayman** [comment]
@smarterclayton Hey, could you please close/decline this PR as the issue was resolved with #35745.

Kind Regards,

Dilyan

**Pull request closed without merging**